README.md

```diff
@@ -146,33 +142,126 @@ There are two main steps in the pre-processing pipeline: creating the graph and
+    source_xy_coordinates
+    surface_geopotential
+    border_mask
+   end
+
+   subgraph dynamic_variables["dynamic variables"]
+    direction LR
+    nwp_mbr_files
+    wtr_files
+    nwp_toa_files
+  end
+
+  dynamic_variables -- read by --> create_parameter_weights_step
+  create_parameter_weights_step -- writes --> parameter_weights
+
+  surface_geopotential -- read by --> create_grid_features_step
+  source_xy_coordinates -- read by --> create_grid_features_step
+  border_mask -- read by --> create_grid_features_step
+  create_grid_features_step -- writes --> grid_features
+
+  subgraph parameter_weights
+    direction LR
+    parameter_mean.pt:::pt_file
+    parameter_std.pt:::pt_file
+    diff_mean.pt:::pt_file
+    diff_std.pt:::pt_file
+    flux_stats.pt:::pt_file
+  end
+end
+subgraph Graph
+  source_xy_coordinates -- read by --> create_mesh_step
+  create_mesh_step -- writes --> mesh_information
+
+  subgraph mesh_information["Mesh information"]
+    direction LR
+    subgraph mesh_common["Mesh (common)"]
+        mesh_features.pt:::pt_file
+        m2m_edge_index.pt:::pt_file
+        g2m_edge_index.pt:::pt_file
+        m2g_edge_index.pt:::pt_file
+        m2m_features.pt:::pt_file
+        g2m_features.pt:::pt_file
+        m2g_features.pt:::pt_file
+    end
+      subgraph mesh_hierarchical["Mesh (hierarchical)"]
+        mesh_down_edge_index.pt:::pt_file
+        mesh_up_edge_index.pt:::pt_file
+        mesh_down_features.pt:::pt_file
+        mesh_up_features.pt:::pt_file
+      end
+  end
+end
+```
 
 ### Create graph
-Run `python -m neural_lam.create_mesh` with suitable options to generate the graph you want to use (see `python neural_lam.create_mesh --help` for a list of options).
+Run `python -m neural_lam.create_graph` with suitable options to generate the graph you want to use (see `python -m neural_lam.create_graph --help` for a list of options).
 The graphs used for the different models in the [paper](https://arxiv.org/abs/2309.17370) can be created as:
 
-* **GC-LAM**: `python -m neural_lam.create_mesh --graph multiscale`
-* **Hi-LAM**: `python -m neural_lam.create_mesh --graph hierarchical --hierarchical` (also works for Hi-LAM-Parallel)
-* **L1-LAM**: `python -m neural_lam.create_mesh --graph 1level --levels 1`
+* **GC-LAM**: `python -m neural_lam.create_graph --config CONFIG --name multiscale`
+* **Hi-LAM**: `python -m neural_lam.create_graph --config CONFIG --name hierarchical --hierarchical` (also works for Hi-LAM-Parallel)
+* **L1-LAM**: `python -m neural_lam.create_graph --config CONFIG --name 1level --levels 1`
 
-The graph-related files are stored in a directory called `graphs`.
+The graph-related files are stored in the same folder as the configuration file under the directory `graph`
 
 ### Create remaining static features
 To create the remaining static files run `python -m neural_lam.create_grid_features` and `python -m neural_lam.create_parameter_weights`.
```

Comment:
Is this completely obsolete now and to be removed?

Reply:
The files output from these are still used in the NpyFilesDatastore, so not completely obsolete. But it would be good to move these in under the directory for that datastore to make this more clear. For the readme this description would fit under the NpyFilesDatastore section.